src/modules/web3/services/EventsQuerier.ts

```diff
@@ -0,0 +1,93 @@
+import { MerkleDistributor, SpokePool } from "@across-protocol/contracts-v2";
+import { TypedEvent, TypedEventFilter } from "@across-protocol/contracts-v2/dist/typechain/common";
+import { Logger } from "@nestjs/common";
+import { Web3Error, Web3ErrorCode } from "../model/ChainId";
+
+const DEFAULT_BLOCK_RANGE = 100_000;
+
+export class EventsQuerier {
+  constructor(
+    private contract: SpokePool | MerkleDistributor,
+    private logger: Logger,
+    private blockRangeSize?: number,
+  ) {}
+
+  public async getEvents(
+    from: number,
+    to: number,
+    filters: TypedEventFilter<TypedEvent<any>[], any>,
+  ): Promise<TypedEvent<any>[]> {
+    let events: TypedEvent<any>[] = [];
+    let retryWithLowerBatchSize;
+
+    do {
+      const blockRangeSizeAtStart = this.blockRangeSize;
+      try {
+        retryWithLowerBatchSize = false;
+        events = [];
+
+        if (this.blockRangeSize) {
+          const intervals = this.getSamplesBetween(from, to, this.blockRangeSize);
+          // query events only for the first interval to make sure block range is fine
+          const [intervalStart, intervalEnd] = intervals[0];
+          const newEvents = await this.contract.queryFilter(filters, intervalStart, intervalEnd);
+          events.push(...newEvents);
+
+          // query the rest of block intervals in parallel in order to get the events
+          const newEventsList = await Promise.all(
+            intervals
+              .slice(1)
+              .map(([intervalStart, intervalEnd]) => this.contract.queryFilter(filters, intervalStart, intervalEnd)),
+          );
+          events.push(...newEventsList.flat());
+        } else {
+          const newEvents = await this.contract.queryFilter(filters, from, to);
+          events.push(...newEvents);
+        }
+      } catch (error) {
+        if (
+          (error as Web3Error).error?.code === Web3ErrorCode.BLOCK_RANGE_TOO_LARGE ||
+          (error as Web3Error).error?.code === Web3ErrorCode.EXCEEDED_MAXIMUM_BLOCK_RANGE ||
+          (error as Web3Error).error?.code === Web3ErrorCode.LOG_RESPONSE_SIZE_EXCEEDED ||
+          (error as Web3Error).error?.code === Web3ErrorCode.LOG_RESPONSE_SIZE_EXCEEDED_2
+        ) {
+          // make sure the block range size wasn't modified by a parallel function call
+          if (this.blockRangeSize === blockRangeSizeAtStart) {
+            const newBlockRangeSize = this.blockRangeSize ? this.blockRangeSize / 2 : DEFAULT_BLOCK_RANGE;
+            this.logger.warn(`lowering block range size from ${this.blockRangeSize} to ${newBlockRangeSize}`);
+            this.blockRangeSize = newBlockRangeSize;
+          }
```

Comment:
From what I see this logic has not been modified, only moved, Right? I just want to double check because I don't see any difference and this is a crucial part used to query the events

Reply:
Yes, exactly. This is only copied and not modified.